### PR TITLE
#162660515 Setup logs for application

### PR DESCRIPTION
**Pull request by frostyblok**
#### What does this PR do?
Setup logs folder and takes record of events when the application runs

#### Description of Task to be completed?
* Set up log folder
* Create a function that logs events for every database error

#### How should this be manually tested?
* Clone repo
* Run `npm install` to install all the dependencies
* Start the development server using `npm run dev:start`.
* Test endpoints with postman for database, then the application log folder gets updated with the details of the error

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories? (If applicable)
[#162660515](https://www.pivotaltracker.com/story/show/162660515)
#### Screenshots (If applicable)

screen shot 2018-12-14 at 1 21 54 pm

### CODE REVIEW

**Comment by okoroemeka** [comment]
@frostyblok you might want to check out your pull request again. You have the last part of your commit message at the top of the pull request description.

**Review by akhilome — COMMENTED** [review]
@frostyblok, hat tip for taking the initiative to complete this task. 👏 

One thing, though:

While reviewing, I noticed that this **exact** block of code:

```js
catch (error) {
    logTracker(error);
    response.status(500).json({
       status: 'Fail',
      message: 'Could not complete action at this time'
  });
}
```

... showed up in multiple locations. Can't we DRY up the codebase further by abstracting this somewhere?

File: src/controllers/article.js [hunk 1]
```diff
@@ -3,6 +3,7 @@ import ArticleModel from '../helpers/articles';
 import { articleAverageRating } from './articleRatingController';
 import TagHelpers from '../helpers/tagHelpers';
 import FavoriteModelHelper from '../helpers/favorite';
+import logTracker from '../../logger/logTracer';
```
Comment (luqmanoop) [inline src/controllers/article.js]: @frostyblok Good job. I would love to know why you've imported this as `logTracker` when the module name is `logTracer` or is this a typo?

**Review by luqmanoop — CHANGES_REQUESTED** [review]
@frostyblok I can't tell you how much I love this work of yours. The logger will be our single source of truth for debugging and getting to the root cause of bugs/problems faster, both locally and in production.

However, though, different things can go wrong in our app, some will need to be fixed immediately, while others may be just warnings or info. We don't want to be logging every one of these things together, there should be some kind of a way we can log based on the severity of what went wrong in our app. I should be able to log something as an error or info or warning. Please check this out https://github.com/winstonjs/winston#logging and make changes where appropriate

File: src/controllers/article.js (hunk 1 above)
Comment (frostyblok) [reply src/controllers/article.js]: This was completely a typographical error. I am going to make the necessary changes ASAP. Thank you for spotting this out.

Concerning your other feedback, I am happy with the fact you took time to go through the PR. But I believe I have something exactly like that in my implementation. For instance, every time we start the server, the logger logs it as an 'info', and when there is an error in the catch block, the logger logs it as 'error'

This is the logger in the server.js
screen shot 2018-12-19 at 9 57 41 am

And this is the logger we have in the catch block
screen shot 2018-12-19 at 9 58 07 am

**Review by luqmanoop — APPROVED** [review]
@frostyblok Good to go.

**Review by akhilome — APPROVED** [review]

File: src/controllers/HighLightedText.js [hunk 2]
```diff
@@ -1,8 +1,10 @@
 import models from '../db/models';
 import ArticleModel from '../helpers/articles';
 import errorResponse from '../helpers';
+import logTracker from '../../logger/logTraker';
 
 const { HighlightedText: HighLightedTextModel } = models;
+const customError = 'Could not complete action at this time';
```
Comment (omokaro-faith) [inline src/controllers/HighLightedText.js]: rename to errorMessage?

File: src/controllers/article.js [hunk 3]
```diff
@@ -4,6 +4,9 @@ import { articleAverageRating } from './ratingController';
 import TagHelpers from '../helpers/tagHelpers';
 import FavoriteModelHelper from '../helpers/favorite';
 import { hasReadArticle } from './statisticsController';
+import logTracker from '../../logger/logTraker';
+
+const customError = 'Could not complete action at this time';
```
Comment (omokaro-faith) [inline src/controllers/article.js]: here also and the files that follow this.

**Pull request merged**